### ISSUE TITLE
macOS API Documentation

### ISSUE THREAD

**Issue by z3t0** [koekeishiya/chunkwm#87]
Hi! I am really interested in the development of alternative windowing systems in macOS but have always hit walls in my own projects since there are no publicly documented APIs. 

Is there some place where one can find out more about the APIs used to create chunkwm

**Comment by koekeishiya** [comment]
Most of the documents I've been reading are legacy references. A lof of the stuff written do not apply anymore and so you have to be able to filter and look for the important things that are still available today.

The publicly documented API's that I use in chunkwm is more or less the following:

NSWorkspace (SharedWorkspace ??)
Carbon
CoreGraphics
ApplicationServices
Probably other stuff, 

See here https://developer.apple.com/documentation/applicationservices

I don't remember where I found my other stuff, but I have saved some pdfs.

In short, you need curiosity and persistence to keep trying and experimenting.
There is not a whole lot of docs out there in this area.
